source/geod24/LocalRest.d

```diff
@@ -390,131 +655,175 @@ public final class RemoteAPI (API) : API
+                                break;
+
+                            case MessageType.response :
+                                auto c = thisScheduler.newCondition();
+                                foreach (_; 0..10)
+                                {
+                                    if (thisLocalWaitingManager.exist(msg.res.id))
+                                        break;
+                                    c.wait(10.msecs);
+                                }
+                                if (!isSleeping())
+                                    handleRes(msg.res);
+                                else if (!control.drop)
+                                    await_res ~= msg.res;
+                                break;
+
+                            case MessageType.filter :
+                                control.filter = msg.filter;
+                                break;
+
+                            case MessageType.time_command :
+                                control.sleep_until = Clock.currTime + msg.time.dur;
+                                control.drop = msg.time.drop;
+                                control.send_response_msg = msg.time.send_response_msg;
+                                break;
+
+                            case MessageType.shutdown_command :
+                                terminate = true;
+                                throw new OwnerTerminated();
+
+                            default :
+                                assert(0, "Unexpected type: " ~ msg.tag);
+                        }
+                    }
+
+                    if (thisScheduler !is null)
+                        thisScheduler.yield();
 
-                    // now handle any leftover messages after any sleep() call
                     if (!isSleeping())
                     {
-                        await_msgs.each!(msg => msg.tag == 0 ? handle(msg.res) : handle(msg.cmd));
-                        await_msgs.length = 0;
-                        assumeSafeAppend(await_msgs);
+                        if (await_req.length > 0)
+                        {
+                            await_req.each!(cmd => handleCmd(cmd));
+                            await_req.length = 0;
+                            assumeSafeAppend(await_req);
+                        }
+
+                        if (await_res.length > 0)
+                        {
+                            await_res.each!(res => handleRes(res));
+                            await_res.length = 0;
+                            assumeSafeAppend(await_res);
+                        }
+
+                        if (await_drop_req.length > 0)
+                        {
+                            await_drop_req.each!(cmd => handleDropCmd(cmd));
+                            await_drop_req.length = 0;
+                            assumeSafeAppend(await_drop_req);
+                        }
                     }
+
```

Comment:
This can never be `false`

Reply:
I tried many things to solve the exception. So this is how it happened.
I'll revise it. 😀